keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationSupport.kt

```diff
@@ -46,13 +46,26 @@ class AuthorizationSupport(
 
   fun userCanModifySpec(serviceAccount: String, specOrName: Any): Boolean {
     val auth = SecurityContextHolder.getContext().authentication
-    return userCanAccessServiceAccount(auth, serviceAccount, specOrName)
+    return userCanAccessServiceAccount(auth, serviceAccount, "Resource: $specOrName")
+  }
+
+  fun userCanModifyApplication(application: String): Boolean = try {
+    val deliveryConfig = repository.getDeliveryConfigsByApplication(application).first()
+    userCanModifyDeliveryConfig(deliveryConfig.serviceAccount, application, deliveryConfig.name)
+  } catch (e: NoSuchElementException) {
+    // If there are no delivery configs for that application return true so this is handled correctly in the controller.
+    true
+  }
+
+  fun userCanModifyDeliveryConfig(serviceAccount: String, application: String, manifestName: String): Boolean {
+    val auth = SecurityContextHolder.getContext().authentication
+    return userCanAccessServiceAccount(auth, serviceAccount, "Delivery config for application $application: $manifestName")
   }
```

Comment:
You can simplify this and make it useful for other use cases like so:
```suggestion
  fun userCanModifyDeliveryConfig(deliveryConfig: DeliveryConfig): Boolean {
    val auth = SecurityContextHolder.getContext().authentication
    return userCanAccessServiceAccount(
      auth,
      deliveryConfig.serviceAccount,
      "Delivery config for application ${deliveryConfig.application}: ${deliveryConfig.name}"
    )
  }
```